Helpjuice/ckeditor5
# [Editor] [sc-2100] [Improvement] Add option to insert PDF preview directly in the Article Editor

## Describe the changes in this pull request

This PR adds a new component to the editor: `EmbeddedIFrame`. It will be used when PDFs are inserted from the Files Manager. It shows the URL preview, allows the user to resize it, and it allows the user to transform it to a link.

App PR: https://github.com/hajrice/helpjuice4/pull/2981

### Demo:

https://github.com/Helpjuice/ckeditor5/assets/28270050/0ad14a22-bb0f-4a8a-893b-8c65faab267a

### Does this resolve an existing issue?

[[sc-2100](https://app.shortcut.com/helpjuice/story/2100)]

### How should this be tested during the review?

Try to insert the following HTML through the source editor on the editor preview:

```
<iframe style="height:500px;width:100%;" src="https://s3.amazonaws.com/helpjuice-dev/helpjuice_development/uploads/upload/image/14128/2692701/1685014732996-1684737927841-sample.pdf" frameborder="0" data-embedded-iframe="true"></iframe>
```

The editor parses it and shows the `<iframe>`. If the iframe is selected, a toolbar appears with inputs for specifying the height and width, and a button that transforms the `<iframe>` to a link (if you click it and wants to go back to an `<iframe>` press `Ctrl+Z`).
